### PR TITLE
add note about httproute precedence and cleanup

### PR DESCRIPTION
**Pull request by wmorgan**
Signed-off-by: William Morgan <william@buoyant.io>

### CODE REVIEW

File: linkerd.io/content/2.12/reference/authorization-policy.md [hunk 1]
```diff
@@ -130,6 +142,13 @@ authorizing traffic to that `HTTPRoute` only rather than to the entire [Server].
 `HTTPRoutes` may also define filters which add processing steps that must be
 completed during the request or response lifecycle.
 
+{{< note >}}
+A given HTTP request can only match on HTTPRoute. If multiple HTTPRoutes
```
Comment (hawkw) [inline linkerd.io/content/2.12/reference/authorization-policy.md:146]: typo: 
```suggestion
A given HTTP request can only match one HTTPRoute. If multiple HTTPRoutes
```

File: linkerd.io/content/2.12/reference/authorization-policy.md [hunk 2]
```diff
@@ -50,9 +61,10 @@ pod/port pairs. Linkerd ships with an admission controller that prevents
 overlapping `Server`s from being created.
 
 {{< note >}}
-When a Server selects a port, all traffic to that port is then denied by
-default, regardless of the default policy. Thus, to authorize traffic to a port
-selected by a Server, you must create AuthorizationPolicies.
+When a Server resource is present, all traffic to the port on its pods will be
+denied (regardless of the default policy) unless explicitly authorized. Thus,
+Servers are typically paired with e.g. an AuthorizationPolicy that references
+the Server, or that reference an HTTPRoute that in turn references the Server.
```
Comment (hawkw) [inline linkerd.io/content/2.12/reference/authorization-policy.md:67]: nit: we use code formatting for the names of kube resources in the body text of this doc, but not in the notes --- should we make this more consistent?
```suggestion
When a `Server` resource is present, all traffic to the port on its pods will be
denied (regardless of the default policy) unless explicitly authorized. Thus,
`Servers` are typically paired with e.g. an `AuthorizationPolicy` that references
the `Server`, or that reference an `HTTPRoute` that in turn references the `Server`.
```

File: linkerd.io/content/2.12/reference/authorization-policy.md (hunk 1 above)
Comment (hawkw) [inline linkerd.io/content/2.12/reference/authorization-policy.md:146]:
```suggestion
A given HTTP request can only match on `HTTPRoute`. If multiple `HTTPRoute`s
```

File: linkerd.io/content/2.12/reference/authorization-policy.md (hunk 2 above)
Comment (wmorgan) [reply linkerd.io/content/2.12/reference/authorization-policy.md:67]: Just seemed a bit overkill...

**Pull request merged**